### DISCUSSION TITLE
Error when install and run in ubuntu 22.04

### DISCUSSION THREAD

**Discussion by nguyenloi85** [selldone/storefront#5] [Q&A] [answered]
I'm using ubuntu 22.04 lts. Yesterday, I cloned and tried to install and run storefront. But I got some errors with both installation and run. 
The first thing is that I can't run command `npm i`. I got these errors bellow
![Screenshot from 2024-01-09 23-24-28](https://github.com/selldone/storefront/assets/13583925/d4156f72-957c-4b7d-9fda-0e3755e17b2a)
Then I tried to skip these error by adding flag `--legacy-peer-deps` in `npm i` command. The installation process  was passed but I can't run 'npm install'  command like the guide in github. I have to run ./install.sh command
Afterwards, I tried to run it, but the app has lots of errors like theses. I can't even count how many errors the apps has
![Screenshot from 2024-01-09 23-30-17](https://github.com/selldone/storefront/assets/13583925/3fc6ea27-b281-40d0-9efb-c0e1f6a5c2ff)
 The  web app can't  be started. when I access the url, the web app shows empty blank page
 Can you please help me resolve these errors abve
 Thanks in advance.

**Comment by pajuhaan** [comment, marked as answer]
I've checked with '**npm**' and found that the issue was caused by '**npm**', since we usually work with '**yarn**'. I've implemented a fix to ensure compatibility with 'npm'. **Please update and try running it with npm now**.

Here are the steps for running in development mode:

1. Run 'setup.sh': This will add modules (we prefer 'modules' over 'library' as the project is in the ALPHA stage, allowing developers complete control over all code parts within the same project in one IDE).
2. Run 'yarn install' or 'npm install' to download the necessary dependencies.
3. Run 'yarn serve' or 'npm serve': This will read the '.env' file, build a development version of the project, and create a local server at 'localhost:8080'.
4. Open 'https://localhost:8080/' (you'll need to manually accept the SSL self-certificate, or disable serving over HTTPS, which is not recommended as you'll be working with real Selldone data, and most functionalities won't work over HTTP on the client side).

You should see your project with our sample store data. If you want to load your shop data, simply change the **.env file**. If you encounter any difficulties, please let me know here.

Plz ignore these on run 'npm serve': (all TS errors will be fixed in future updates)
![image](https://github.com/selldone/storefront/assets/4279061/68264554-4f3c-45db-b246-8bd26d8dbeca)

**Reply by nguyenloi85** [reply]
@pajuhaan actually I tried with both npm and yarn before. After you answered me, I tried with both npm and yarn again. I can run npm install or yarn install without errors. But when I tried to run it via browser with address `https://localhost:8080`. It got the same errors as it had before. The website shows the empty blank page, with coutless error in terminal, and some api errors in console tab of browser. Please check more details in the image below. 
![image](https://github.com/selldone/storefront/assets/13583925/e0761b3d-d461-4f5e-a6e5-fe9cf75924c6)
Can you tell me more about your local development?
which operation system you are using? linux, windows or macos?
which version of yarn are you using?
I will tried to install and used the same as yours to save time

**Reply by pajuhaan** [reply]
The problem was because of the missing **.env file** in the project. I pushed the sample .env. 


Link: https://github.com/selldone/storefront/blob/main/.env

![image](https://github.com/selldone/storefront/assets/4279061/6e4a6e84-f9ff-4e2c-8f24-832fa173421b)

**Comment by pajuhaan** [comment]
The problem is **.env** file!
![image](https://github.com/selldone/storefront/assets/4279061/e3980828-9f59-4640-8e0e-15a710f3980c)

You need to set the shop name in the env to load data:
![image](https://github.com/selldone/storefront/assets/4279061/459a84cb-5c45-4c47-8db8-2bd7ee37bd0d)


Replace this dor sample '.env':

```
# ▃▃▃▃▃▃▃▃▃▃▃▃ Sample Storefront ▃▃▃▃▃▃▃▃▃▃▃▃

VUE_APP_SAMPLE_SHOP_NAME=toysworld

#Number: custom landing page number
VUE_APP_SAMPLE_SHOP_CUSTOM_HOME=3001

VUE_APP_SAMPLE_DEFAULT_CURRENCY=GBP
VUE_APP_SAMPLE_DEFAULT_COUNTRY=GB
VUE_APP_SAMPLE_DEFAULT_LANGUAGE=en


VUE_APP_SAMPLE_THEME_COLOR_LIGHT="#70557E"
VUE_APP_SAMPLE_THEME_COLOR_DARK="#000000"
VUE_APP_SAMPLE_THEME_COLOR_DEEP_DARK="#000000"
VUE_APP_SAMPLE_THEME_COLOR_INFO="#E91E63"

# ▃▃▃▃▃▃▃▃▃▃▃▃ Dev Server ▃▃▃▃▃▃▃▃▃▃▃▃

VUE_DEV_SERVER_HOST=
VUE_DEV_SERVER_PORT=
VUE_DEV_SERVER_HTTPS=TRUE
```
